app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,72 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export_id)
+    ActiveRecord::Base.connection_pool.with_connection do
+      csv_export = CsvExport.find(csv_export_id)
+      create_export_folder(csv_export)
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # clean up downloaded files older than 12 hours or any jobs stuck for over 1 month
+    # TODO delete files with no record associations
+    end_date = Time.now - Rails.application.config.samson.export_job.downloaded_age
+    timeout_date = Time.now - Rails.application.config.samson.export_job.max_age
+    @csv_exports = CsvExport.where("(status = 'downloaded' AND updated_at <= :end_date) OR updated_at <= :timeout_date",
+      {end_date: end_date, timeout_date: timeout_date})
+    @csv_exports.each do |csv_export|
+      csv_export.destroy
+      Rails.logger.info("Downloaded file #{csv_export.download_name} deleted")
```

Comment:
Moving the logging info into the delete_file function.  That way it'll state if it succeeds or fails.